### PR TITLE
WB-1310: Refactor SearchTextInput to be sticky

### PR DESCRIPTION
**Pull request by jandrade**
## Summary:

- Moved `SearchField` out of the listbox (DropdownCore).
- Removed `SearchTextInput` component.
- Cleaned up some code that is no longer needed due to these changes.
- Updated tests to reflect the change.

Issue: WB-1310

## Test plan:

Open `Dropdown > SingleSelect` and `Dropdown > MultiSelect` and verify that the
search field works fine and the list items display correctly.

https://5e1bf4b385e3fb0020b7073c-esqmrhemky.chromatic.com/?path=/story/dropdown-singleselect--virtualized-filterable

https://user-images.githubusercontent.com/843075/171708391-88a689f0-ff94-4814-8866-7f31996d28de.mov

### CODE REVIEW

File: packages/wonder-blocks-dropdown/src/components/dropdown-core.js [hunk 1]
```diff
@@ -419,34 +426,55 @@ class DropdownCore extends React.Component<Props, State> {
     }
 
     focusCurrentItem() {
-        const fousedItemRef = this.state.itemRefs[this.focusedIndex];
+        const focusedItemRef = this.state.itemRefs[this.focusedIndex];
```
Comment (jandrade) [inline packages/wonder-blocks-dropdown/src/components/dropdown-core.js:429]: Just made a renaming here to fix a typo.

File: packages/wonder-blocks-dropdown/src/components/dropdown-core.js [hunk 2]
```diff
@@ -664,30 +686,6 @@ class DropdownCore extends React.Component<Props, State> {
                 ? this.state.itemRefs[focusIndex].ref
                 : null;
 
-            // Render the SearchField component.
```
Comment (jandrade) [inline packages/wonder-blocks-dropdown/src/components/dropdown-core.js:667]: I've simplified a lot how the `SearchField` is rendered (now it's completely separate from the `listbox` logic.

File: packages/wonder-blocks-dropdown/src/components/dropdown-core.js [hunk 3]
```diff
@@ -774,6 +748,32 @@ class DropdownCore extends React.Component<Props, State> {
...
+                clearAriaLabel={labels.clearSearch}
+                onChange={this.handleSearchTextChanged}
+                placeholder={labels.filter}
+                ref={this.searchFieldRef}
+                style={styles.searchInputStyle}
+                value={searchText || ""}
+            />
+        );
+    }
+
```
Comment (jandrade) [inline packages/wonder-blocks-dropdown/src/components/dropdown-core.js:776]: I moved "SearchTextInput" here by removing `search-text-input.js` altogether and simplifying this by calling the WB's `SearchField`s component directly.

File: packages/wonder-blocks-dropdown/src/components/dropdown-core.js [hunk 4]
```diff
@@ -788,33 +788,34 @@ class DropdownCore extends React.Component<Props, State> {
...
-                        maxDropdownHeight,
-                    ),
-
                     dropdownStyle,
                 ]}
+                testId="dropdown-core-container"
             >
-                {listRenderer}
+                {this.props.isFilterable && this.renderSearchField()}
+                <View
```
Comment (jandrade) [inline packages/wonder-blocks-dropdown/src/components/dropdown-core.js:807]: Created this new wrapper to contain the `OptionItems` and separate the search field from that container.

BEFORE: 
```
<div role="listbox">
  <search />
  <option />
  <option />
  ...
```

AFTER
```
<div container>
  <search />
  <div role="listbox">
    <option />
    <option />
    ...
```

As an extra, this is now more semantically correct as we don't include a `textbox` inside a `listbox`.

File: packages/wonder-blocks-dropdown/src/components/dropdown-core.js [hunk 5]
```diff
@@ -916,6 +918,14 @@ const styles = StyleSheet.create({
         marginTop: Spacing.xxSmall_6,
     },
 
+    searchInputStyle: {
```
Comment (jandrade) [inline packages/wonder-blocks-dropdown/src/components/dropdown-core.js:921]: Moved the search styles directly here.

**Review by jeresig — APPROVED** [review]
This looks much better, thank you! It looks like there was a noticeable drop in code coverage - is that something that can be fixed?

**Comment by jandrade** [comment]
> It looks like there was a noticeable drop in code coverage - is that something that can be fixed?

That's like an issue with codecov... I checked and the coverage was increased by `0.04%` but the chart looks like there was a drop. You can see the changes here:

https://app.codecov.io/gh/Khan/wonder-blocks/compare/1516/tree/packages/wonder-blocks-dropdown/src/components

I'm not sure if that happened because I got rid of `search-text-input.js`.

P.S. I noticed that I forgot to add a changeset so I'll add that before merging.

**Comment by jandrade** [comment]
The parent pull-request ([#1513](https://github.com/Khan/wonder-blocks/pull/1513)) has been merged into `main`, but this branch (`dd-sticky`) now has conflicts with the new base branch. These conflicts must be resolved before checks can complete on this pull-request.

**Pull request merged**